Source/Promise.swift

```diff
@@ -107,7 +129,7 @@ extension Future {
         return future
     }
     
-    func flatMap<Q>(_ block:@escaping (T)->(Q?)) -> Future<Q> {
+    public func flatMap<Q>(_ block:@escaping (T)->(Q?)) -> Future<Q> {
```

Comment:
This is just `map`. The parameter of `flatMap` should be `(T) -> Future<Q>`

Reply:
But the block is executed asynchronously. It couldn't immediately return like that. But ok. I'll change the name.